src/snowcli/cli/common/flags.py

```diff
@@ -1,28 +1,35 @@
 from __future__ import annotations
 
-from typing import Optional
+from typing import Any, Callable, Optional
 
 import typer
-from snowcli.cli.common.snow_cli_global_context import (
-    ConnectionDetails,
-    update_callback,
-)
+from snowcli.cli.common.cli_global_context import cli_context_manager
 from snowcli.output.formats import OutputFormat
 
 DEFAULT_CONTEXT_SETTINGS = {"help_option_names": ["--help", "-h"]}
 
-
 _CONNECTION_SECTION = "Connection configuration"
 _CLI_BEHAVIOUR = "Global configuration"
 
 
+def _callback(provide_setter: Callable[[], Callable[[Any], Any]]):
+    def callback(value):
+        set_value = provide_setter()
+        set_value(value)
+        return value
+
+    return callback
+
+
 ConnectionOption = typer.Option(
     None,
     "--connection",
     "-c",
     "--environment",
     help=f"Name of the connection, as defined in your `config.toml`. Default: `dev`.",
-    callback=ConnectionDetails.update_callback("connection_name"),
+    callback=_callback(
+        lambda: cli_context_manager.connection_context.set_connection_name
+    ),
```

Comment:
It won't work correctly because it will use connection_context instance from the moment of flags initialization which can be different (and it is different) than connection_context from the moment of callback execution.

However I can make `ConnectionContext` to be a singleton and add `reset` method to it.

Reply:
But IMO lambda should be used here because we shouldn't depend on implementation of cli_context or connection_context.